From 20014427c5b225ca47a546f0d77aab981e9ee849 Mon Sep 17 00:00:00 2001
From: Jakub Filak <jfilak@redhat.com>
Date: Tue, 4 Nov 2014 09:14:43 +0100
Subject: [LIBREPORT PATCH 114/118] Revert "ureport: consistently die on all
 client auth errors"

This reverts commit fca34b5620a6213e1d191bcb99e8757f34bbe0a5.

Related #1140224

Signed-off-by: Jakub Filak <jfilak@redhat.com>
---
 src/lib/ureport.c | 28 +++++++++++++++++++++++++---
 1 file changed, 25 insertions(+), 3 deletions(-)

diff --git a/src/lib/ureport.c b/src/lib/ureport.c
index ae6a58d..fef3922 100644
--- a/src/lib/ureport.c
+++ b/src/lib/ureport.c
@@ -119,12 +119,21 @@ ureport_server_config_set_client_auth(struct ureport_server_config *config,
             ureport_server_config_set_url(config, xstrdup(RHSM_WEB_SERVICE_URL));
 
         char *rhsm_dir = rhsm_config_get_entitlement_cert_dir();
+        if (rhsm_dir == NULL)
+        {
+            log_notice("Not using client authentication");
+            return;
+        }
 
         GList *certs = get_file_list(rhsm_dir, "pem");
         if (g_list_length(certs) < 2)
         {
             g_list_free_full(certs, (GDestroyNotify)free_file_obj);
-            error_msg_and_die("'%s' does not contain a cert-key files pair", rhsm_dir);
+
+            log_notice("'%s' does not contain a cert-key files pair", rhsm_dir);
+            log_notice("Not using client authentication");
+            free(rhsm_dir);
+            return;
         }
 
         /* Use the last non-key file found. */
@@ -141,7 +150,11 @@ ureport_server_config_set_client_auth(struct ureport_server_config *config,
         if (cert == NULL)
         {
             g_list_free_full(certs, (GDestroyNotify)free_file_obj);
-            error_msg_and_die("'%s' does not contain a cert file (only keys)", rhsm_dir);
+
+            log_notice("'%s' does not contain a cert file (only keys)", rhsm_dir);
+            log_notice("Not using client authentication");
+            free(rhsm_dir);
+            return;
         }
 
         config->ur_client_cert = xstrdup(fo_get_fullpath(cert));
@@ -177,7 +190,16 @@ ureport_server_config_set_client_auth(struct ureport_server_config *config,
                         xasprintf(RHSMENT_SIG_DATA_BEGIN_TAG"%s"RHSMENT_SIG_DATA_END_TAG, sig_data));
             }
             else
-                error_msg_and_die("Cert file '%s' doesn't contain Entitlement and RSA Signature sections", config->ur_client_cert);
+            {
+                log_notice("Cert file '%s' doesn't contain Entitlement and RSA Signature sections", config->ur_client_cert);
+                log_notice("Not using client authentication");
+
+                free(config->ur_client_cert);
+                config->ur_client_cert = NULL;
+
+                free(config->ur_client_key);
+                config->ur_client_key = NULL;
+            }
 
             free(sig_data);
             free(ent_data);
-- 
1.8.3.1

